docs/DeveloperGuide.md

```diff
@@ -304,11 +304,106 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. SoConnect shows an error message.
 
       Use case resumes at step 2.
 
-*{More to be added}*
+**Use case: Search a contact*
+
+**MSS**
+
+1.  User requests to search a specific word in the list
+2.  SoConnect shows a list of persons related to the word.
+
+**Extension**
+* 2a. The list is empty.
+
+  Use case ends
+
+* 3a. There is no input after search.
+* 
+    * 3a1. SoConnect shows the same list of persons.
+  
+      Use case ends.
+
+**Use case: Edit a tag**
+
+**MSS**
+
+1.  User requests for list of tags.
+2.  SoConnect shows the list of tags.
+3.  User changes a tag.
+4.  SoConnect updates the tag.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. There are no tags.
+
+  Use case ends.
+
+* 3a. There is no such current tag.
+    * 1a1. SoConnect shows an error message.
+    
+      Use case resumes at step 2
+* 3b. The new tag already exist.
+    * 3b1. SoConnect shows an error message.
+    
+      Use case resumes at step 2.
+
+**Use case: Add tag to Contact**
+
+**MSS**
+
+1.  User requests for the name of the contact.
+2.  SoConnect gives the contact with the right name.
+3.  User request to add tag to the contact.
+4.  SoConnect adds the tag to the contact.
+
+    Use case ends.
+
+**Extensions**
+
+*  1a. There is no such name in the contacts.
+    * 1a1. SoConnect shows an error message.
+      Use case resumes at step 1.
+*  3a. There is no such tag in the taglist.
+    * 3a1. SoConnect shows aan error message.
+      Use case resumes at step 2.
+
+**Use case: Autocomplete a word**
+
+**MSS**
+
+1.  User inputs word
+2.  SoConnect gives a selection of possible words.
+3.  User chooses the right word.
+4.  SoConnect changes inputted word to the chosen word. 
+
+    Use case ends.
+
+**Extension**
+*  1a. The word changes.
+    * 1a1. SoConnect updates the selection of possible words.
```

Comment:
After each extension, it should be stated how the use case progresses.

For example for this extension, it should be `Use case resumes from step 3.`